### DISCUSSION TITLE
How to reschedule the pods to a different node if the pod on this specific node is not coming to Running state?

### DISCUSSION THREAD

**Discussion by reddybhavaniprasad** [kubernetes/website#46225] [General]
Hi,

I have deployed **Mongodb** as a **replicaset** on a RKE2 cluster with 3 replicas. One of them acts as **PRIMARY** node and the other 2 replicas act as **SECONDARY** nodes.
Unfortunately, one of the worker nodes on which the PRIMARY node is scheduled is taken for maintenance then the replica on this VM is trying to come up until the maintenance window is finished.
Here the worker node is still in READY state, so the pod is not rescheduled on a different node. Also, the pod did not come into READY state until the maintenance is finished.

Please help me in understating the reason for why the PRIMARY mongo replica is not rescheduled on a difference working node. Also help me if I can update the manifests in order to reschedule it to another node in this case.

Mongodb Version: 4.4.12
Replicaset with 3 replicas (1 PRIMARY & 2 SECONDARY & 1 ARBITER)

**Cluster information:**
```
Kubernetes version: 1.23
Cloud being used: Bare-Metal VMs created from VMWare
Installation method: Air Gapped RKE2 Installation
Host OS: RHEL 8.6
CNI and version: calico - v3.22.1
CRI and version: crictl - v1.23.0
3 - Control Nodes (54GB RAM and 24 core CPU)
15 - Compute Nodes (54GB RAM and 24 core CPU)
```

**Comment by tengqm** [comment]
/kind support
This doesn't look like a docs issue. Please try stackoverflow or related Kubernetes slack channel.

**Comment by sftim** [comment]
GitHub is not the right place for support requests.

If you're looking for help, check [Server Fault](https://serverfault.com/questions/tagged/kubernetes).

You can also post your question on the [Kubernetes Slack](http://slack.k8s.io/) or the [Discuss Kubernetes](https://discuss.kubernetes.io/) forum.
